code/game/gamemodes/cult/ritual.dm

```diff
@@ -171,26 +171,16 @@ This file contains the arcane tome files.
 	var/list/shields = list()
 	var/area/A = get_area(src)
 
-	if(locate(/obj/effect/rune) in Turf)
-		user << "<span class='cult'>There is already a rune here.</span>"
-		return
-
-	if(Turf.z != ZLEVEL_STATION && Turf.z != ZLEVEL_MINING)
-		user << "<span class='warning'>The veil is not weak enough here."
+	if(!check_rune_turf(Turf, user))
 		return
-	if(istype(A, /area/shuttle))
-		user << "<span class='warning'>Interference from hyperspace engines disrupts the Geometer's power on shuttles.</span>"
 	for(var/T in subtypesof(/obj/effect/rune) - /obj/effect/rune/malformed)
 		var/obj/effect/rune/R = T
 		if(initial(R.cultist_name))
 			possible_runes.Add(initial(R.cultist_name)) //This is to allow the menu to let cultists select runes by name rather than by object path. I don't know a better way to do this
 	if(!possible_runes.len)
 		return
 	entered_rune_name = input(user, "Choose a rite to scribe.", "Sigils of Power") as null|anything in possible_runes
-	if(!Adjacent(user) || !src || qdeleted(src) || user.incapacitated())
-		return
-	if(isspaceturf(Turf))
-		user << "<span class='warning'>You cannot scribe runes in space!</span>"
+	if(!Adjacent(user) || !src || qdeleted(src) || user.incapacitated() || !check_rune_turf(Turf, user))
```

Comment:
It isn't, as far as I can tell; it'll return FALSE if you give it something that's null.